modules/javafx.graphics/src/main/java/javafx/scene/Parent.java

```diff
@@ -991,10 +991,13 @@ boolean isPerformingLayout() {
     private double minHeightCache = -1;
 
     void setLayoutFlag(LayoutFlags flag) {
+        // Needs to be set before needsLayout is updated, as otherwise a listener that
+        // calls isNeedsLayout() might see the old value.
+        layoutFlag = flag;
```

Comment:
This is the right change, but I suspect it might cause regression.

The JavaFX entities which call `isNeedsLayout()`:

`VirtualFlow::layoutChildren`
`VirtualFlow::setCellIndex`

We may need to focus on List/Table/Tree/TableViews during testing.

(Anecdata: I've seen continuous layout calls in the TableView before)

Reply:
It's very unlikely to cause regressions, because `needsLayoutProperty()` is not used anywhere in JavaFX. The only thing that is different now is that a listener added to this property will see the correct value when calling `isNeedsLayout()`.

Reply:
you are probably right, re:javafx.  applications might listen to this property, though it's rather unlikely.

Reply:
Yes, but I don't think there's anything we can do about it short of not fixing the bug...